src/client/components/ToggleSwitchInput.tsx

```diff
@@ -0,0 +1,213 @@
+  width: 2.75rem;
+  height: 1.5rem;
+  border-radius: 15.5px;
+
+  /* this will go away when resets have been standardised */
+  &:before,
+  &:after {
+    box-sizing: border-box;
+  }
+
+  &:before {
+    content: '';
+    position: absolute;
+    top: 6px;
+    height: 11px;
+    width: 6px;
+    right: 10px;
+    opacity: 0;
+    border-bottom: 2px solid ${success[400]};
+    border-right: 2px solid ${success[400]};
+    transform: rotate(45deg);
+    transition: opacity 0.1s ease-in;
+  }
+
+  &:after {
+    height: 18px;
+    width: 18px;
+    top: 3px;
+    left: 4px;
+  }
+`;
+
+export type Platform = 'web';
+export type LabelPosition = 'left' | 'right';
+
+export interface ToggleSwitchInputProps extends Props {
+  /**
+   * Whether the ToggleSwitch is checked. This is necessary when using the
+   * [controlled approach](https://reactjs.org/docs/forms.html#controlled-components)
+   * (recommended) to form state management.
+   *
+   * Note: if you pass the `checked` prop, you MUST also pass an `onClick`
+   * handler, or the field will be rendered as read-only.
+   */
+  checked?: boolean;
+  /**
+   * When using the [uncontrolled approach](https://reactjs.org/docs/uncontrolled-components.html),
+   * use defaultChecked to indicate the whether the ToggleSwitch is checked initially.
+   */
+  defaultChecked?: boolean;
+  /**
+   * Optional Id for the switch. Defaults to a generated indexed Source ID e.g. "src-component-XXX}"
+   */
+  id?: string;
+  /**
+   * Appears to the right of the switch by default.
+   */
+  label?: string;
+  /**
+   * Which side of the switch the label will appear on.
+   */
+  labelPosition?: LabelPosition;
+  /**
+   * Sets the toggle styling appropriate for each platform.
+   * The default platform is 'web'.
+   */
+  platform?: Platform;
+}
+
+/**
+ * [Storybook](https://guardian.github.io/source/?path=/docs/kitchen-source-react-components-development-kitchen-toggle-switch--playground) •
+ * [Design System](https://theguardian.design) •
+ * [GitHub](https://github.com/guardian/source/tree/main/packages/@guardian/source-react-components-development-kitchen/components/toggle-switch) •
+ * [NPM](https://www.npmjs.com/package/@guardian/source-react-components-development-kitchen)
+ *
+ * Displays an on/off toggle switch. This toggle has default styling and can be used on android, ios or web.
+ * These styles are driven by the 'platform' prop.
+ * To give it more custom styling cssOverride may be used.
+ */
+const getPlatformStyles = (platform: Platform): SerializedStyles => {
+  switch (platform) {
+    case 'web':
+      return webStyles;
+  }
+};
+
+export const ToggleSwitchInput = ({
+  id,
+  label,
+  labelPosition = 'right',
+  checked,
+  defaultChecked,
+  platform = 'web',
+  cssOverrides,
+}: ToggleSwitchInputProps): EmotionJSX.Element => {
+  const switchName = id ?? generateSourceId();
+  const labelId = descriptionId(switchName);
+
+  const isChecked = (): boolean => {
```

Comment:
I think this adds a lot of unnecessary obfuscation, I believe it can be removed by making the change I suggested below